subsystem/cluster_test.go

```diff
@@ -909,9 +909,9 @@ var _ = Describe("Validate BaseDNSDomain when creating a cluster", func() {
 	}
 	tests := []DNSTest{
 		{
-			It:            "V2RegisterCluster should throw an error. BaseDNSDomain='example', not a valid DNS structure string",
+			It:            "V2RegisterCluster should not throw an error. BaseDNSDomain='example', valid DNS",
 			BaseDNSDomain: "example",
-			ShouldThrow:   true,
+			ShouldThrow:   false,
 		},
 		{
 			It:            "V2RegisterCluster should throw an error. BaseDNSDomain='example.c', Invalid top-level domain name ",
```

Comment:
I am confused here, that's what we discussed offline. Why is `example.c` invalid? Top Level Domain of 1 character is technically a valid one. It is defined in https://www.rfc-editor.org/rfc/rfc952 as `<name>  ::= <let>[*[<let-or-digit-or-hyphen>]<let-or-digit>]`.

Apart from that there is https://datatracker.ietf.org/doc/html/rfc3696#section-2 which states
```
additional rule
that essentially requires that top-level domain names not be all-
numeric
```

So `example.c` is a valid name whereas `example.1` is invalid

Reply:
@mkowalski  For some reason the existing validation (which I did not change as it was not part of the PR requirement strictly speaking) requires it to be a 2 or more letter suffix. I Believe it is because we are aiming for a real world DNS address where the last part will be .com or .au). I am not sure we need to relax this limitation. What do you think?

Reply:
As I said, real world does allow for a suffix that contains only 1 letter. Sadly I cannot answer why the original regex had the requirement of "minimum 2 characters" but this requirement does not come from the standard (i.e. RFC, anything what is not in RFC is someone's implementation, not "standard"). It may be one of the vendor's implementation (e.g. Microsoft has more strict requirements for DNS names), it may be someone's invented rule, I just can't say why we had it in the first place (just like we don't know why we had a limitation that we need a "dot" so we landed with the issue you are fixing)

Anyway, you are fixing what got initially reported so I'm okay keeping it like this till we get a request to soften it again in the future. The change is too small and too irrelevant to keep it hanging forever :)